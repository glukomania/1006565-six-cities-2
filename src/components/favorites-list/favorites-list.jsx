import {connect} from 'react-redux';
import Header from '../header/header';
import FavoriteEmpty from '../favorite-empty/favorite-emty';
import {Operations} from '../../store/reducer';
import FavoriteCity from './components/favorite-city/favorite-city';

class FavoritesList extends React.PureComponent {
  constructor(props) {
    super();
    props.loadFavorites();

    this.favoriteCities = Array.from(props.favorites.reduce((acc, elem) => acc.add(elem.city.name), new Set()));
  }

  render() {
    return (this.props.favorites.length === 0) ? <FavoriteEmpty /> : <div>
      <Header isInner={true}/>

      <main className="page__main page__main--favorites">
        <div className="page__favorites-container container">
          <section className="favorites">
            <h1 className="favorites__title">Saved listing</h1>
            <ul className="favorites__list">
              {this.favoriteCities.map((item, index) => <FavoriteCity key={index} city={item} favorites={this.props.favorites.filter((favoriteItem) => favoriteItem.city.name === item)}/>)}
            </ul>
          </section>
        </div>
      </main>
    </div>;
  }
}

FavoritesList.propTypes = {
  favorites: PropTypes.array.isRequired,
  loadFavorites: PropTypes.func.isRequired,
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  favorites: state.favorites,
});

const mapDispatchToProps = {
  loadFavorites: Operations.loadFavorites,
};

export {FavoritesList};
export default connect(mapStateToProps, mapDispatchToProps)(FavoritesList);
