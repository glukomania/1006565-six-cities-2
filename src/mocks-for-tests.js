// export const offers = [
//   {
//     id: 1,
//     city: {
//       name: `Amsterdam`,
//       location: {
//         latitude: `52.369553943508`,
//         longitude: `4.85309666406198`,
//       }
//     },
//     name: `Beautiful & luxurious apartment at great location`,
//     images: [`img/apartment-02.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/apartment-01.jpg`],
//     price: 80,
//     rating: 80,
//     type: `Private room`,
//     propFeatures: {entire: `Entire place`, bedroom: 3, adults: 3},
//     insideItems: [`Wi-Fi`, `Washing machine`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Kitchen`, `Dishwasher`, `Cabel TV`, `Fridge`],
//     host: {name: `William`, status: `Free`, avatar: `img/avatar-angelina.jpg`},
//     description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
//     location: {
//       latitude: `52.369553943508`,
//       longitude: `4.85309666406198`
//     },
//   },
//   {
//     id: 2,
//     city: {
//       name: `Amsterdam`,
//       location: {
//         latitude: `52.369553943508`,
//         longitude: `4.85309666406198`,
//       }
//     },
//     name: `Beautiful & luxurious apartment at great location`,
//     images: [`img/apartment-02.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/apartment-01.jpg`],
//     price: 80,
//     rating: 80,
//     type: `Private room`,
//     propFeatures: {entire: `Entire place`, bedroom: 3, adults: 3},
//     insideItems: [`Wi-Fi`, `Washing machine`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Kitchen`, `Dishwasher`, `Cabel TV`, `Fridge`],
//     host: {name: `William`, status: `Free`, avatar: `img/avatar-angelina.jpg`},
//     description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
//     location: {
//       latitude: `52.369553943508`,
//       longitude: `4.85309666406198`
//     },
//   },
//   {
//     id: 3,
//     city: {
//       name: `Berlin`,
//       location: {
//         latitude: `55.1`,
//         longitude: `4.1`,
//       }
//     },
//     name: `Beautiful & luxurious apartment at great location`,
//     images: [`img/apartment-02.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/apartment-01.jpg`],
//     price: 80,
//     rating: 80,
//     type: `Private room`,
//     propFeatures: {entire: `Entire place`, bedroom: 3, adults: 3},
//     insideItems: [`Wi-Fi`, `Washing machine`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Kitchen`, `Dishwasher`, `Cabel TV`, `Fridge`],
//     host: {name: `William`, status: `Free`, avatar: `img/avatar-angelina.jpg`},
//     description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
//     location: {
//       latitude: `55.1`,
//       longitude: `4.1`
//     },
//   },
// ];


export const offers = [
  {
    'id': 1,
    'city': {
      name: `Amsterdam`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    'preview_image': `img/1.png`,
    'images': [`img/1.png`, `img/2.png`],
    'title': `Beautiful & luxurious studio at great location`,
    'is_favorite': false,
    'is_premium': false,
    'rating': 4.8,
    'type': `apartment`,
    'bedrooms': 3,
    'max_adults': 4,
    'price': 120,
    'goods': [`Heating`, `Kitchen`, `Cable TV`],
    'host': {
      'id': 3,
      'is_pro': true,
      'name': `Angelina`,
      'avatar_url': `img/1.png`
    },
    'description': `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    'location': {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8
    }
  }
];


export const feedbacks = [
  {
    id: 1,
    user: {
      'id': 4,
      'is_pro': false,
      'name': `Max`,
      'avatar_url': `img/1.png`
    },
    rating: 4,
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:56.569Z`,
  }
];
