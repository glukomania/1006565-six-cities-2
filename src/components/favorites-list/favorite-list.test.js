import renderer from 'react-test-renderer';
import {FavoritesList} from './favorites-list';
import {offers} from '../../mocks-for-tests';

jest.mock(`react-router-dom`, () => ({
  Link: () => null
}));

jest.mock(`./components/favorite-city/favorite-city`, () => `FavoriteCity`);
jest.mock(`../header/header`, () => `Header`);

it(`FavoritesList is displayed correctly`, () => {
  const loadFavorites = jest.fn();

  const props = {
    city: `Amsterdam`,
    favorites: offers,
  };

  const favoriteList = renderer
    .create(<FavoritesList {...props} loadFavorites={loadFavorites}/>)
  .toJSON();

  expect(favoriteList).toMatchSnapshot();
});
