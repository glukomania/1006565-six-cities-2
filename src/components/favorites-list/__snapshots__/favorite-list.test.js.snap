// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesList is displayed correctly 1`] = `
<div>
  <Header
    isInner={true}
  />
  <main
    className="page__main page__main--favorites"
  >
    <div
      className="page__favorites-container container"
    >
      <section
        className="favorites"
      >
        <h1
          className="favorites__title"
        >
          Saved listing
        </h1>
        <ul
          className="favorites__list"
        >
          <FavoriteCity
            city="Amsterdam"
            favorites={
              Array [
                Object {
                  "bedrooms": 3,
                  "city": Object {
                    "location": Object {
                      "latitude": 52.370216,
                      "longitude": 4.895168,
                      "zoom": 10,
                    },
                    "name": "Amsterdam",
                  },
                  "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.",
                  "goods": Array [
                    "Heating",
                    "Kitchen",
                    "Cable TV",
                  ],
                  "host": Object {
                    "avatar_url": "img/1.png",
                    "id": 3,
                    "is_pro": true,
                    "name": "Angelina",
                  },
                  "id": 1,
                  "images": Array [
                    "img/1.png",
                    "img/2.png",
                  ],
                  "is_favorite": false,
                  "is_premium": false,
                  "location": Object {
                    "latitude": 52.35514938496378,
                    "longitude": 4.673877537499948,
                    "zoom": 8,
                  },
                  "max_adults": 4,
                  "preview_image": "img/1.png",
                  "price": 120,
                  "rating": 4.8,
                  "title": "Beautiful & luxurious studio at great location",
                  "type": "apartment",
                },
              ]
            }
          />
        </ul>
      </section>
    </div>
  </main>
</div>
`;
